deps/src/singular.cpp

```diff
@@ -43,6 +62,9 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
 
     Singular.method("siInit", [](const char * path) {
         siInit(const_cast<char *>(path));
+        singular_error = std::string();
+        singular_return = std::string();
+        singular_warning = std::string();
```

Comment:
Shouldn't this simply be `foo.clear()` ?

Reply:
This is the first initialization. But I can just statically initialize them, and delete this here. The strings get cleaned up at the beginning of the `call_interpreter` function anyway.

Reply:
No, it is not the first initialization. The first initialization is performed when the C++ runtime runs  through the list of global constructors.

So all in all, the above three lines are not needed at all, and can be safely removed.